src/backend/base/langflow/components/processing/data_operations.py

```diff
@@ -0,0 +1,436 @@
+
+        return filtered_data
+
+    def multi_filter_data(self) -> Data:
+        """Apply multiple filters to the data."""
+        self.validate_single_data("Filter Values")
+        data_filtered = self.get_normalized_data()
+
+        for filter_key in self.filter_key:
+            if filter_key not in data_filtered:
+                msg = f"Filter key '{filter_key}' not found in data. Available keys: {list(data_filtered.keys())}"
+                raise ValueError(msg)
+
+            if isinstance(data_filtered[filter_key], list):
+                for filter_data in self.filter_values:
+                    filter_value = self.filter_values.get(filter_data)
+                    if filter_value is not None:
+                        data_filtered[filter_key] = self.filter_data(
+                            input_data=data_filtered[filter_key],
+                            filter_key=filter_data,
+                            filter_value=filter_value,
+                            operator=self.operator,
+                        )
+            else:
+                msg = f"Filter key '{filter_key}' is not a list."
+                raise TypeError(msg)
+
+        return Data(**data_filtered)
+
+    def append_update(self) -> Data:
+        """Append / Update  with new key-value pairs."""
+        self.validate_single_data("Append / Update ")
+        data_filtered = self.get_normalized_data()
+
+        for key, value in self.append_update_data.items():
+            data_filtered[key] = value
+
+        return Data(**data_filtered)
+
+    # Configuration and execution methods
+    def update_build_config(self, build_config: dotdict, field_value: Any, field_name: str | None = None) -> dotdict:
+        """Update build configuration based on selected action."""
+        if field_name != "operations":
+            return build_config
+
+        self.operations = field_value
+        selected_actions = [action["name"] for action in self.operations]
+
+        # Handle single action case
+        if len(selected_actions) == 1 and selected_actions[0] in ACTION_CONFIG:
+            action = selected_actions[0]
+            config = ACTION_CONFIG[action]
+
+            build_config["data"]["is_list"] = config["is_list"]
+            logger.info(config["log_msg"])
+
+            return set_current_fields(
+                build_config=build_config,
+                action_fields=self.actions_data,
+                selected_action=action,
+                default_fields=self.default_keys,
+            )
+
+        # Handle no operations case
+        if not selected_actions:
+            logger.info("setting default fields")
+            return set_current_fields(
+                build_config=build_config,
+                action_fields=self.actions_data,
+                selected_action=None,
+                default_fields=self.default_keys,
+            )
+
+        return build_config
+
+    def as_data(self) -> Data:
```

Comment:
Do we want this to return Data for sure, rather than a DataFrame? (I'm still not clear on our preference as a team...)